apps/web/src/components/Composer/OpenActions.tsx

```diff
@@ -1,17 +1,81 @@
-import type { UnknownOpenActionModuleSettings } from '@hey/lens';
-import type { FC } from 'react';
-
 import SwapOpenAction from '@components/Publication/OpenAction/UnknownModule/Swap';
+import Oembed from '@components/Shared/Oembed';
 import { XMarkIcon } from '@heroicons/react/24/outline';
+import { KNOWN_ATTRIBUTES } from '@hey/data/constants';
 import { VerifiedOpenActionModules } from '@hey/data/verified-openaction-modules';
+import getURLs from '@hey/helpers/getURLs';
+import { type UnknownOpenActionModuleSettings } from '@hey/lens';
+import { MetadataAttributeType } from '@lens-protocol/metadata';
+import { type FC, useEffect } from 'react';
 import { useOpenActionStore } from 'src/store/non-persisted/publication/useOpenActionStore';
+import { usePublicationAttachmentStore } from 'src/store/non-persisted/publication/usePublicationAttachmentStore';
+import { usePublicationAttributesStore } from 'src/store/non-persisted/publication/usePublicationAttributesStore';
+import { usePublicationStore } from 'src/store/non-persisted/publication/usePublicationStore';
+
+type OpenActionsProps = {
+  nftOpenActionEmbed?: boolean;
+  nftOpenActionEmbedLoading?: boolean;
+};
 
-const OpenActions: FC = () => {
+const OpenActions: FC<OpenActionsProps> = ({
+  nftOpenActionEmbed,
+  nftOpenActionEmbedLoading
+}) => {
   const { openAction, reset } = useOpenActionStore();
 
+  const { publicationContent, quotedPublication } = usePublicationStore();
+  const { attachments } = usePublicationAttachmentStore((state) => state);
+  const { addAttribute, getAttribute, removeAttribute } =
+    usePublicationAttributesStore();
+
+  const urls = getURLs(publicationContent);
+
+  useEffect(() => {
+    if (urls.length) {
+      removeAttribute(KNOWN_ATTRIBUTES.HIDE_OEMBED);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [urls.length]);
+
   const hasSwapOpenAction =
     openAction?.address === VerifiedOpenActionModules.Swap;
 
+  if (
+    !urls.length ||
+    attachments.length ||
+    quotedPublication ||
+    getAttribute(KNOWN_ATTRIBUTES.HIDE_OEMBED)?.value === 'true'
+  ) {
+    return null;
+  }
+
+  if (Boolean(nftOpenActionEmbed)) {
+    return (
+      <div className="relative m-5">
+        <Oembed
+          nftOpenActionEmbed={nftOpenActionEmbed}
+          nftOpenActionEmbedLoading={nftOpenActionEmbedLoading}
+          url={urls[0]}
+        />
```

Comment:
Again as earlier this should live in separate component and not use oembed

Oembed => should be only used for link previews
OpenAction => should have individual set of components for each open actions

Please refer swap or rent OA and how it got consumed and structured